File: docs/packages/hw1/hw1/src/robot.cpp
#include <iostream>

#include <cstdlib>
#include <ctime>

#include <ros/ros.h>
#include "hw1/robot_position.h"
#include "hw1/velocities.h"

using namespace std;
using namespace ros;

void get_velocities (const hw1::velocities::ConstPtr& vel){
    float v = vel->Forward_Vel;
    float w = vel->Forward_Vel;
    ROS_INFO("Subscribing to Velocities topic as : [%f, %f]",v,w);
}
   
int main (int argc, char **argv){

    int freq = 50;
    int max_size = 100;

    init(argc,argv,"Robot");

    NodeHandle n;

    Publisher pub = n.advertise<hw1::robot_position> ("Position", max_size);
    Subscriber sub = n.subscribe("Velocities", max_size, get_velocities);

    Rate rate(freq);

    srand(static_cast<unsigned int>(time(0)));  

    while (ok()){
        hw1::robot_position pos;

        pos.xR = static_cast<float>(rand()) / RAND_MAX * 100.0f;
        pos.yR = static_cast<float>(rand()) / RAND_MAX * 100.0f;
        pos.thR = static_cast<float>(rand()) / RAND_MAX * 3.14f;
        
        pub.publish(pos);
        spinOnce();
        rate.sleep();
    }
 
    return 0;
}

File: packages/hw1/hw1/src/Robot.py
import rospy
from hw1.msg import robot_position, velocities
import numpy as np

def get_velocities (vel):
    print('Subscribing to Velocities topic')

rospy.init_node('Robot')
rospy.Subscriber('Velocities', velocities, get_velocities)
pub = rospy.Publisher('Position', robot_position, queue_size=100)
rate = rospy.Rate(50)

while not rospy.is_shutdown():
    pos = robot_position()
    pos.xR = 100*np.random.rand()
    pos.yR = 100*np.random.rand()
    pos.thR= np.random.rand()
    pub.publish(pos)
    rate.sleep()

